cmd/calyptia/delete_core_instance.go

```diff
@@ -0,0 +1,166 @@
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func newCmdDeleteCoreInstance(config *config, testClientSet kubernetes.Interface) *cobra.Command {
+	var skipError bool
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	cmd := &cobra.Command{
+		Use:               "core_instance CORE_INSTANCE",
+		Aliases:           []string{"instance", "aggregator"},
+		Short:             "Delete a core instance",
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: config.completeAggregators,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			aggregatorKey := args[0]
+			var agg types.Aggregator
+			var err error
+			if validUUID(aggregatorKey) {
+				agg, err = config.cloud.Aggregator(ctx, aggregatorKey)
+				if err != nil {
+					return err
+				}
+			} else {
+				aggregatorID, err := config.loadAggregatorID(aggregatorKey)
+				if err != nil {
+					return err
+				}
+				agg, err = config.cloud.Aggregator(ctx, aggregatorID)
+				if err != nil {
+					return err
+				}
+			}
+
+			if configOverrides.Context.Namespace == "" {
+				configOverrides.Context.Namespace = apiv1.NamespaceDefault
+			}
+
+			var clientset kubernetes.Interface
+			if testClientSet != nil {
+				clientset = testClientSet
+			} else {
+				kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+				kubeClientConfig, err := kubeConfig.ClientConfig()
+				if err != nil {
+					return err
+				}
+
+				clientset, err = kubernetes.NewForConfig(kubeClientConfig)
+				if err != nil {
+					return err
+				}
+
+			}
+
+			k8sClient := &k8s.Client{
+				Interface:    clientset,
+				Namespace:    configOverrides.Context.Namespace,
+				ProjectToken: config.projectToken,
+				CloudBaseURL: config.baseURL,
+			}
+
+			if err := k8sClient.EnsureOwnNamespace(ctx); err != nil {
+				return fmt.Errorf("could not ensure kubernetes namespace exists: %w", err)
+			}
+
+			label := "aggregatorID=" + agg.ID
+
+			deployments, err := k8sClient.FindDeploymentsByLabel(ctx, label)
+			if err != nil {
+				return err
+			}
+
+			for _, item := range deployments.Items {
+				err = k8sClient.DeleteDeployment(ctx, item.Name)
+				if err != nil {
+					if !skipError {
+						return err
+					}
+					fmt.Fprintf(cmd.OutOrStdout(), "%q was not deleted, err: %v\n", item.Name, err)
+				}
+				fmt.Fprintf(cmd.OutOrStdout(), "deployment=%q deleted\n", item.Name)
+			}
+
+			services, err := k8sClient.FindServicesByLabel(ctx, label)
+			if err != nil {
+				return err
+			}
+
+			for _, item := range services.Items {
+				err := k8sClient.DeleteService(ctx, item.Name)
+				if err != nil {
+					if !skipError {
+						return err
+					}
+					fmt.Fprintf(cmd.OutOrStdout(), "%q was not deleted, err: %v\n", item.Name, err)
+				}
+				fmt.Fprintf(cmd.OutOrStdout(), "service=%q deleted\n", item.Name)
+			}
+
+			clusterRole := agg.Name + "-cluster-role"
```

Comment:
Shouldn't all resources be deleted by labels?

Reply:
i  changed all for delete by label, but maintained the code for logging :)